packages/api/src/utils/billing.js

```diff
@@ -1,24 +1,40 @@
 /* eslint-disable no-void */
 
+import { InvalidTosAgreementError, TosAgreementRequiredError } from '../../src/errors.js'
+
 /**
  * @typedef {import('./billing-types').StoragePriceName} StoragePriceName
  */
 
+/**
+ * @param {import('./billing-types').Agreement} agreement
+ * @returns {boolean}
+ */
+function isAgreement (agreement) {
+  return agreement === 'web3.storage-tos-v1'
+}
+
 /**
  * Save a user's payment settings
  * @param {object} ctx
  * @param {import('./billing-types').BillingService} ctx.billing
  * @param {import('./billing-types').CustomersService} ctx.customers
  * @param {import('./billing-types').SubscriptionsService} ctx.subscriptions
+ * @param {import('./billing-types').TermsOfServiceService} ctx.termsOfService
  * @param {import('./billing-types').BillingUser} ctx.user
  * @param {object} paymentSettings
  * @param {Pick<import('./billing-types').PaymentMethod, 'id'>} paymentSettings.paymentMethod
  * @param {import('./billing-types').W3PlatformSubscription} paymentSettings.subscription
+ * @param {import('./billing-types').Agreement} paymentSettings.agreement
  * @param {import('./billing-types').UserCreationOptions} [userCreationOptions]
  */
 export async function savePaymentSettings (ctx, paymentSettings, userCreationOptions) {
-  const { billing, customers, user } = ctx
+  if (!paymentSettings.agreement) throw new TosAgreementRequiredError()
```

Comment:
I *think* we only need to require this when the user is actually saving their paymentSettings to be a subscription.storage value that isn't `null`, and maybe not even when it's `free`. Not exactly sure, it's a business/legal call but this is the function that is meant to encode that 'business logic' (which we should test well once we understand it to prevent regressions).

So if `paymentSettings.subscription.storage === null` we probably don't need to require an agreement.